pkg/services/ngalert/state/manager.go

```diff
@@ -175,6 +175,14 @@ func (st *Manager) setNextState(alertRule *ngModels.AlertRule, result eval.Resul
 	case eval.Pending: // we do not emit results with this state
 	}
 
+	// Set Resolved property so the scheduler knows to send a postable alert
+	// to alert manager.
+	if oldState == eval.Alerting && currentState.State == eval.Normal {
```

Comment:
Apologies as I'm not really familiar with all of the states, but I'd assume this should be:

```
oldState == eval.Alerting && currentState.State != eval.Alerting
```

Unless there's no chance we can go from alerting -> pending? (e.g. in the case of a restart)

Reply:
I think we don't sent notifications on Error / or NoData states (@davidmparrott ?) So I don't transitions to those states would be recoveries.

Reply:
Correct, notifications are not sent for NoData or Error states. Users can configure the NoData state to resolve to Alerting if they want to get notified, though.